### ISSUE TITLE
Cannot update selection tag in aws_backup_selection

### ISSUE THREAD

**Issue by awilliamsOM1** [terraform-providers/terraform-provider-aws#8443]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v0.11.11
provider.aws v2.6.0

### Affected Resource(s)

* aws_backup_selection

### Terraform Configuration Files

```hcl
resource "aws_backup_selection" "standard_backup_selection" {
  plan_id      = "${aws_backup_plan.standard_backup_plan.id}"
  name         = "standard_backup_selection"
  iam_role_arn = "..."

  selection_tag {
    type  = "STRINGEQUALS"
    key   = "NEW_TAG"
    value = "NEW_VALUE"
  }
}
```

### Debug Output

```
Terraform will perform the following actions:

  ~ module.backup_vault.aws_backup_selection.standard_backup_selection
      selection_tag.4271826135.key:   "" => "backup.plan"
      selection_tag.4271826135.type:  "" => "STRINGEQUALS"
      selection_tag.4271826135.value: "" => "standard"
      selection_tag.706126364.key:    "backup_plan" => ""
      selection_tag.706126364.type:   "STRINGEQUALS" => ""
      selection_tag.706126364.value:  "standard" => ""


Plan: 0 to add, 1 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

module.backup_vault.aws_backup_selection.standard_backup_selection: Modifying... (ID: 8e3c89b1-50e7-4750-98b5-e2300151af31)
  selection_tag.4271826135.key:   "" => "backup.plan"
  selection_tag.4271826135.type:  "" => "STRINGEQUALS"
  selection_tag.4271826135.value: "" => "standard"
  selection_tag.706126364.key:    "backup_plan" => ""
  selection_tag.706126364.type:   "STRINGEQUALS" => ""
  selection_tag.706126364.value:  "standard" => ""

Error: Error applying plan:

1 error(s) occurred:

* module.backup_vault.aws_backup_selection.standard_backup_selection: 1 error(s) occurred:

* aws_backup_selection.standard_backup_selection: doesn't support update
```

### Expected Behavior

I changed the tag used as the selection for backing up a resource using a backup plan. aws_backup_selection do not support updates, so what should have happened is the old aws_backup_selection should have been deleted and a new resource created.

### Actual Behavior

Terraform plan shows the tag in the aws_backup_selection will be changed, but terraform apply fails because `aws_backup_selection.standard_backup_selection: doesn't support update`

### Steps to Reproduce

1. Create an aws_backup_vault, aws_backup_plan, and a aws_backup_selection where the selection is selecting using a tag
2. `terraform apply`
3. Change the aws_backup_selection tag
4. `terraform apply` again to get the error.

### Important Factoids

None

### References

* None

**Comment by awilliamsOM1** [comment]
My workaround was:
1. Delete the `aws_backup_selection`
2. `terraform apply`
3. Recreate the `aws_backup_selection` with the updated tag
4. `terraform apply`

**Comment by bflad** [comment]
The fix to properly trigger resource recreation with `selection_tag` updates in the `aws_backup_selection` resource has been merged and will release with version 2.11.0 of the Terraform AWS Provider, likely tomorrow. 👍

**Comment by bflad** [comment]
This has been released in [version 2.11.0 of the Terraform AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#2110-may-17-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
